### PR TITLE
modify the UML diagram according to our discussion today

### PR DESCRIPTION
**Pull request by Taranjot-Singh**
Hello Friends,

Please see this UML diagram and provide any thoughts.

In particular, these are things to note/discuss:

1) After removing the ViewModels for the Fragments, all the methods that handle the logic for what happens when a button on the fragment is clicked are now in the parent activity's ViewModel class. We will have to "propogate" the button clicks back to the parent activity so we can access the methods of the parent's ViewModel but that should not be too hard and the methods for that seem irrelevant to include here.

2) I don't quite understand why we need the aggregation between the ProfileFragment class and the ProfileActivity class. I get that they are similar so shouldn't we just keep one or the other? Like just keep the ProfileActivity and have the BookFragement directly pull-up the ProfileActivity when a user clicks on a requester to view their profile?

3) Currently, any button click that should modify the status of a book in the library or modify anything related to the books in the library is handled by the LibraryViewModel and likewise for the UnlibraryViewModel. The handoff seems very similar in the sense that it modifies the status of a book for both the borrower and the owner so I put two functions in the LibraryViewModel and UnlibraryViewModel for handling handoff. I don't think we need a new activity.

4) The ProfileFragment stores a Profile Object.  Where does that come from? I am assuming we store all requesters in the Book object when the status of the book is "requested" and that is where it comes from. Also, what is the Profile class?

Made some other minor changes for consistency

image

### CODE REVIEW

File: doc/p1-uml.txt [hunk 1]
```diff
@@ -44,58 +57,54 @@ class GoogleMapService {
...
+
+class UnlibraryRepository {
 }
 
-class ProfileFragmentViewModel {
-- Profile mProfile
-+ void sendApproval()
-+ void sendDecline()
+class UnlibraryViewModel {
+- ArrayList<Book> oBooks
```
Comment (cyrusdiego) [inline doc/p1-uml.txt:77]: Nit: typo should be mBooks

**Review by cyrusdiego — CHANGES_REQUESTED** [review]
>I don't quite understand why we need the aggregation between the ProfileFragment class and the ProfileActivity class. I get that they are similar so shouldn't we just keep one or the other? Like just keep the ProfileActivity and have the BookFragement directly pull-up the ProfileActivity when a user clicks on a requester to view their profile?

They are similar, the fragment basically just hosts the metadata that we'd pass to it from the Profile class. The ProfileActivity just represents the "Profile" Tab from the navbar and the ProfileActivity would hold a ProfileFragment with the current user's meta-data. 

>The ProfileFragment stores a Profile Object. Where does that come from? I am assuming we store all requesters in the Book object when the status of the book is "requested" and that is where it comes from. Also, what is the Profile class?

There may have been confusion regarding the Profile object, it should actually be called the User object. I believe the origins of the User object depends on the path. For instance, if we go to ProfileActivity, it would be from ProfileViewModel which gets the data from the ProfileRepository. For the UnLibrary and Exchange, it would be from their respective view models and repositories. The user class is part of the Model in MVVM, it represents the owner of the book for instance (name, email, etc.)

**Review by cyrusdiego — APPROVED** [review]

**Comment by Taranjot-Singh** [comment]
I do see a problem here then.  When an owner sees the list of requesters in the detailed book view and clicks on one, where does the User object for that requester come from? Because the LibraryViewModel only holds a list of owner's own books and those Book object only store the owner's User object as of now and not the User objects of every requester: https://github.com/CMPUT301F20T24/Unlibrary/pull/23  

We could modify the Book Class to store an array of references to all requesters. Then this would make sense.

**Comment by armiantos** [comment]
Enforcing Caleb's data model #19, I would imagine when we click the BookFragment specialized under Library, we would filter the requests table that match the current book's ID. And it's from here we can pass the user data to the ProfileFragment. 

I personally think that Activities should be dummies. They are entry points that can be activated at any point.

File: doc/p1-uml.txt [hunk 2]
```diff
@@ -1,36 +1,49 @@
...
-+LibraryViewModel()
+- ArrayList<Book> mBooks
++ LibraryViewModel()
++ void setTitle(String newTitle)
++ void setDescription(String newDescription)
++ void setISBN(String newISBN)
++ void takePictures()
++ void sendApproval()
++ void sendDecline()
++ perfromBorrowHandoff(Book book)
```
Comment (armiantos) [inline doc/p1-uml.txt:45]:
```suggestion
+ initiateLendHandoff(Book book)
```

File: doc/p1-uml.txt [hunk 3]
```diff
@@ -1,36 +1,49 @@
...
++ performReturnHandoff(Book book)
```
Comment (armiantos) [inline doc/p1-uml.txt:46]:
```suggestion
+ inititateReturnHandoff(Book book)
```

**Review by armiantos — DISMISSED** [review]
Left some minor improvements. Overall, great job Taran!

**Comment by Taranjot-Singh** [comment]
image

**Review by cyrusdiego — APPROVED** [review]
LGTM! Double-check Armi's comments are resolved too, I think they are.

**Review by ghassanzadeh — APPROVED** [review]
LGTM

**Pull request merged**